cmake/KokkosCore_config.h.in

```diff
@@ -41,6 +40,8 @@
 #cmakedefine KOKKOS_ENABLE_HIP_RELOCATABLE_DEVICE_CODE
 #cmakedefine KOKKOS_ENABLE_HIP_MULTIPLE_KERNEL_INSTANTIATIONS
 #cmakedefine KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY
+#cmakedefine KOKKOS_ENABLE_IMPL_SYCL_RELOCATABLE_DEVICE_CODE
+#cmakedefine KOKKOS_IMPL_SYCL_DEVICE_GLOBAL_SUPPORTED
```

Comment:
What is this, why is it not KOKKOS_ENABLE?

Reply:
Are these two options orthogonal?

Reply:
> What is this, why is it not KOKKOS_ENABLE?

It's not an option but we detect support at configure-time. I moved it here since it seemed to be a better fit as backend-specific macro definitions.

Reply:
> Are these two options orthogonal?

Not entirely. When we have proper support for global device variables (the feature macro is defined) we can use either implementation in desul and we prefer the non-rdc one right now if we can. If the feature macro is not defined, we need to use the rdc implementation and so we check for KOKKOS_ENABLE_IMPL_SYCL_RELOCATABLE_DEVICE_CODE before defining KOKKOS_IMPL_SYCL_DEVICE_GLOBAL_SUPPORTED in that case.